mailpoet/lib/Automation/Engine/Control/TriggerHandler.php

```diff
@@ -82,7 +82,13 @@ public function processTrigger(Trigger $trigger, array $subjects): void {
       $automationRun = new AutomationRun($automation->getId(), $automation->getVersionId(), $trigger->getKey(), $subjects);
       $stepRunArgs = new StepRunArgs($automation, $automationRun, $step, $subjectEntries, 1);
 
-      if (!$this->filterHandler->matchesFilters($stepRunArgs)) {
+      $match = false;
+      try {
+        $match = $this->filterHandler->matchesFilters($stepRunArgs);
+      } catch (Exceptions\Exception $e) {
+        ;
```

Comment:
Let's maybe put a comment here? Sth like `// failed filter evaluation won't match`. Should we catch `Throwable` or do you think it's better to be more specific?

Reply:
https://github.com/mailpoet/mailpoet/pull/5197/commits/df75a23569e929c3f7c76b66e961c9a2ec15dc78 for the comment. Regarding the scope, I do not have a strong opinion. My thinking was to use a specific error class as the ones under `Exceptions\Exception` are basically `Exceptions::fieldLoadFailed` and `Exceptions::fieldNotFound` for which we defined: In this context that means `false`. Other throwables could be bugs which would hide in the dark with this `catch()`.